gsoc-2020/extension-activity-monitor.md

```diff
@@ -0,0 +1,53 @@
+# Extension Activity Monitor 
+
+**Mentor:** Luca Greco, Andreas Wagner
+
+**Email:** lgreco@mozilla.com, awagner@mozilla.com
+
+## Project Description
+
+Extensions do most of their work invisibly from users, and so  Extension activity is a complete mystery for most users, even advanced ones.
+
+Providing more transparency could help increase reliability of abuse reporting and accountability for developers, as well as providing an
+additional useful tool to aid investigating bugs in the extensions or in the Firefox WebExtensions internals.
+
+## Skills Required
+
+An applicant needs familiarity with:
+
+* HTML/JavaScript and the Web API
+* Revision Control Systems (e.g. git and/or mercurial)
+
+and the following skills:
+
+* Contributing to Open Source
+* Think critically
+* Fluent in english and good reading comprehension
+* Do independent research
+* Enjoy working with others
+* Be productive with little supervision
+* Able to work remotely
+
+"Bonus" skills:
+
+* Familiarity with the Browser Extensions APIs
+
+## Project Details
+
+An `activityLog` API is available (in Firefox >= 70) to privileged extensions, the applicant's goal is leveraging this API to
+create a privileged extension which would explore various ways to present this information to the users.
+
+In particular, this project goal is to expore the following areas:
+
+- Browser tab (about:extensionmonitor, maybe) that shows all extensions active in all open tabs, extensions altering search, and for each installed extension
+  show activity like network requests and data interactions between content scripts and background scripts.
```

Comment:
"between content scripts and background scripts" are extension-specific terminology that may not be understood by everyone.

How about:

"between the different scripts within the extension"

or

"between the different parts of the extension"